src/Symfony/Component/DependencyInjection/Loader/FileLoader.php

```diff
@@ -242,6 +241,17 @@ private function findClasses(string $namespace, string $pattern, array $excludeP
             }
         }
 
+        if (null !== $prefixLen) {
+            foreach ($excludePaths as $path => $_) {
+                $class = $namespace.ltrim(str_replace('/', '\\', substr($path, $prefixLen, str_ends_with($path, '.php') ? -4 : null)), '\\');
+                if (!$this->container->has($class)) {
```

Comment:
this might cause issues when using file loaders in bundle extensions: at that point, they work with a dedicated ContainerBuilder that does not have the services of other bundles.
I think you need to update the `ContainerBuilder::merge()` method to make it aware of the `container.excluded` tag (forbidding an excluded definition to override a non-excluded one during merging)

Reply:
Good call. PR updated, now tested.